reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Retry.kt

```diff
@@ -0,0 +1,33 @@
+package com.badoo.reaktive.completable
+
+import com.badoo.reaktive.base.operator.Retry
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+import kotlin.reflect.KClass
+
+fun Completable.retry(predicate: (attempt: Int, Throwable) -> Boolean = { _, _ -> true }): Completable =
+    completable { emitter ->
+        val disposableWrapper = DisposableWrapper()
+        emitter.setDisposable(disposableWrapper)
+
+        subscribeSafe(
+            object : CompletableObserver, CompletableCallbacks by emitter {
+                private val retry = Retry(emitter, predicate)
+
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onError(error: Throwable) {
+                    retry.onError(error) { this@retry.subscribeSafe(this) }
+                }
+            }
+        )
+    }
+
+fun Completable.retry(times: Int): Completable =
+    retry { attempt, _ -> attempt < times }
+
+fun Completable.retry(throwableType: KClass<out Throwable>): Completable =
+    retry { _, throwable -> throwableType.isInstance(throwable) }
```

Comment:
Isn't `isInstance` requires reflection that is not available on Native?

Reply:
Covered with tests. iOS looks fine, CI will check Linux.

Reply:
I have checked, it works on Linux. I'm still thinking about inline reified variant, it would be very nice to have.

I see two options:
1. remove default value for `predicate` argument
2. rename the last variant to `retryOnException` or similar

I would better go with the first option